src/commands/GameCommand.ts

```diff
@@ -1,17 +1,51 @@
 import { ICommand } from "./ICommand";
+import { BusColor } from "../model/BusColor";
 
 export class GameCommand implements ICommand{
+
     type:string
     data:Object
     privateData:Object
     player:string
     id:number
+    message:string
     constructor(type:string, data:Object, privateData:Object, player:string, id:number){
         this.type=type
         this.data=data
         this.privateData=privateData
         this.player=player
         this.id =id
+        this.message=this.generateMessage();
+    }
+    generateMessage(){
+        if(this.type==="drawBus"){
+            return this.drawBus();
+        } else if(this.type ==="discardRoutes"){
+            return this.drawRoutes();
+        } else if(this.type ==="incrementTurn"){
+            return this.incrementTurn();
+        } else{
+            return "unknown command?!? "+this.player +" "+this.type+" "+this.data;
+        }
+    }
+    drawBus(){
+        let card_type:BusColor=this.data as BusColor;
+        if(this.data==""){
+            return this.player + " drew a card from the top."
+        }
+        return this.player + " drew a "+ card_type.toString + " card."
+    }
+    drawRoutes(){
+        let temp= this.data as {"numCardsKept":number};
+        let numCards=temp.numCardsKept;
+        if(numCards===1)
```

Comment:
An option for the pluralizing could be this:
return this.player + " drew " + String(numCards) + " route card" + (numCards === 1 ? "." : "s.");

I haven't tested that, but that's the idea.